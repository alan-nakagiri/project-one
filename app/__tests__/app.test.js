import React from 'react'
import { render } from '@testing-library/react-native'
import App, { buildArticle } from '../index.js'

describe('the home screen', () => {
	test('renders no articles', () => {
		const { toJSON } = render(<App />)

		expect(toJSON()).toMatchSnapshot()
	})

	// add test to mock state or api
	// test selecting a category
})

describe('buildArticle', () => {
	test('when given articles with fields and sys', () => {
		const article = buildArticle({ fields: { title: 'title', minutes: '10', eyebrowText: 'test' }, sys: { id: 1 } })

		expect(article).toEqual({
			title: 'title',
			minutes: '10',
			id: 1,
			category: 'test'
		})
	})
	// add other edge cases
})

describe('addArticleToCategory', () => {

})

describe('parseArticles', () => {

})