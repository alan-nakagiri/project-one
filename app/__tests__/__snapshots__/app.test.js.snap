// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the home screen renders no articles 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "paddingBottom": 20,
    }
  }
>
  <View
    style={
      {
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontWeight": "bold",
          "marginLeft": 8,
        }
      }
    >
      Categories:
    </Text>
    <RCTScrollView
      contentContainerStyle={
        {
          "padding": 8,
        }
      }
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={
        {
          "flexGrow": 0,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "backgroundColor": "white",
        "paddingHorizontal": 20,
        "paddingVertical": 12,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
